includes/datavalues/SMW_DV_Number.php

```diff
@@ -71,10 +71,20 @@ static protected function parseNumberValue( $value, &$number, &$unit ) {
 		$decseparator = NumberFormatter::getInstance()->getDecimalSeparatorForContentLanguage();
 		$kiloseparator = NumberFormatter::getInstance()->getThousandsSeparatorForContentLanguage();
 
-		$parts = preg_split( '/([-+]?\s*\d+(?:\\' . $kiloseparator . '\d\d\d)*' .
-		                      '(?:\\' . $decseparator . '\d+)?\s*(?:[eE][-+]?\d+)?)/u',
-		                      trim( str_replace( array( '&nbsp;', '&#160;', '&thinsp;', ' ' ), '', $value ) ),
-		                      2, PREG_SPLIT_DELIM_CAPTURE );
+		// #753
+		$regex = '/([-+]?\s*(?:' .
+				// Either numbers like 10,000.99 that start with a digit
+				'\d+(?:\\' . $kiloseparator . '\d\d\d)*(?:\\' . $decseparator . '\d+)?' .
```

Comment:
If you want to support numbers like "4." and "5.e-5" you can replace `(?:\\' . $decseparator . '\d+)?` with `(?:\\' . $decseparator . '\d*)?`.